.github/workflows/dockerbuild.yml

```diff
@@ -14,51 +14,37 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: crazy-max/ghaction-docker-buildx@v3
-
-      - name: Prepare
-        id: prepare
-        run: |
-          DOCKER_IMAGE=robertslando/zwave2mqtt
-          DOCKER_PLATFORMS=linux/arm64,linux/amd64,linux/arm/v6,linux/arm/v7,linux/386
-
-          echo ::set-output name=docker_image::${DOCKER_IMAGE}
-          echo ::set-output name=docker_platforms::${DOCKER_PLATFORMS} \
-
-      - name: Just build on PR
-        if: ${{ github.event_name == 'pull_request' }}
-        run: |
-          docker buildx build \
-          -f docker/Dockerfile \
-          --platform ${{ steps.prepare.outputs.docker_platforms }} \
-          -t ${{ steps.prepare.outputs.docker_image }}:dev \
-          .
 
+      - uses: docker/setup-buildx-action@v1
+      - uses: docker/setup-qemu-action@v1
       - name: Login to dockerhub
         if: ${{ github.event_name != 'pull_request' }}
-        run: |
-          docker login --username $DOCKER_USERNAME --password $DOCKER_PASSWORD
-        env:
-          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
-          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
 
-      - name: Dev
-        if: ${{ github.ref == 'refs/heads/master'}}
-        run: |
-          docker buildx build \
-          -f docker/Dockerfile \
-          --platform ${{ steps.prepare.outputs.docker_platforms }} \
-          -t ${{ steps.prepare.outputs.docker_image }}:dev \
-          --push \
-          .
-
-      - name: Latest & tag
-        if: ${{ github.event_name == 'release' }}
+      - name: Prepare
         run: |
-          docker buildx build \
-          -f docker/Dockerfile \
-          --platform ${{ steps.prepare.outputs.docker_platforms }} \
-          -t ${{ steps.prepare.outputs.docker_image }}:$(echo ${GITHUB_REF} | sed "s/refs\/tags\/v//") \
-          -t ${{ steps.prepare.outputs.docker_image }}:latest \
-          --push \
-          .
+          DOCKER_REPO=robertslando/zwave2mqtt
+          if [ "$GITHUB_REF" == "refs/heads/master" ]; then
+            TAGS="${DOCKER_REPO}:dev\n"
+          fi
+          if [ "$GITHUB_EVENT_NAME" == "release" ]; then
+            TAGS="${TAGS}${DOCKER_REPO}:latest\n"
+            TAGS="${TAGS}${DOCKER_REPO}:$(echo ${GITHUB_REF} | sed "s/refs\/tags\/v//")\n"
+          fi
+
+          TAGS="${TAGS}${DOCKER_REPO}:sha-${GITHUB_SHA}"
```

Comment:
New tag?

Reply:
Yeah, sorry forgot to call out

Reply:
Is it used for caching?

Reply:
No, more for allowing someone to use something ahead of a release without opening themselves to a breaking change coming in next time they pull

Reply:
And what about the 'just build on pr' part? It will push also pull requests with the format `robertslando/zwave2mqtt:sha-<sha>`?

Reply:
Only prs on this repo, prs from forks don't get the secret to push

Reply:
Oh missed the if in the login part

Reply:
its more that forks don't get access to the secrets, otherwise anyone could just get your dockerhub creds